src/components/UserList.js

```diff
@@ -1,81 +1,45 @@
-
+import { connect } from 'react-redux';
+import { fetchUsers, searchInput, filterUser, sortUser } from '../actions/userAction';
 
 class UserList extends Component {
-   
-    state = {
-        search: '',
-        loading: true,
-        users: []
-    };
-    componentWillReceiveProps(props, state) {
-
-    }
-    componentWillMount() {
 
+    componentDidMount(){
+        this.props.dispatch(fetchUsers());
     }
-    componentDidMount() {
-        this.fetchData();
-    }
-    shouldComponentUpdate(nextProps, nextState) {
-        const { users, search, loading } = this.state;
-        return (
-            (JSON.stringify(users)) !== (JSON.stringify(nextState.users)) || search !== nextState.search ||
-            loading !== nextState.loading
-        )
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        const { search } = this.state;
-        (search !== prevState.search) && this.searchingFromFetchData(search);
-
-    }
-    componentWillUnmount() {
-    }
-    fetchData(search) {
-        let url = `https://api.github.com/users`;
-        if (search) url = `${url}/${search}`;
-        fetch(url)
-            .then(users => {
-                return (users.status === 200) ? users.json() : []
-            })
-            .then(users => {
-                users = (Array.isArray(users)) ? users : [users];
-                this.setState({ users, loading: false })
-            })
-            .catch(err => console.log(err));
-    }
-    searchingFromFetchData(search) {
-        if (!search) return this.fetchData()
-        let users = this.state.users.filter(u => {
-            if (u.login.indexOf(search) > -1) {
-                return u;
-            }
-        })
-        this.setState({ users });
-    }
-
-
-    searchandler = (event) => {
+    searcHandler = (event) => {
         let { nativeEvent: { target: { value } } } = event;
-        this.setState({ search: value })
+        this.props.dispatch(searchInput(value));
+        this.props.dispatch(filterUser())
 
     }
+    sortHandler = () => {
+        this.props.dispatch(sortUser());
+    }
     render() {
-
-        const { users } = this.state;
+        let { users, fetching, tempUsers } = this.props;
         return (
             <div>
                 <h1>Users</h1>
-                <Search search={this.searchandler} />
-                {this.state.loading
+                <Search searcHandler={this.searcHandler} sortHandler={this.sortHandler} />
+                {fetching
                     ? 'Loading...'
-                    : <User users={users} />
+                    : <User users={(tempUsers.length < 1) ? users : tempUsers} />
                 }
             </div>
         )
     }
 }
-export default UserList;
\ No newline at end of file
+const mapStateToProps = function (store) {
+    return {
+        users: store.user.users,
+        fetching: store.user.fetching.dispatch,
+        search: store.user.search,
+        tempUsers: store.user.tempUsers
```

Comment:
Instead of hard-coding the accessed objects here, write selector functions, that will return the required props.

Place them here:
`src/selectors/users-list`

Use `index.js`